airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/java/io/airbyte/integrations/destination/bigquery/formatter/DefaultBigQueryDenormalizedRecordFormatter.java

```diff
@@ -235,10 +237,32 @@ public Schema getBigQuerySchema(final JsonNode jsonSchema) {
     };
   }
 
+  private static JsonNode getFileDefinition(final JsonNode fieldDefinition) {
+    if (fieldDefinition.has(TYPE_FIELD)) {
+      return fieldDefinition;
+    } else {
+      if (fieldDefinition.has(ANY_OF_FIELD) && fieldDefinition.get(ANY_OF_FIELD).isArray()) {
+        var fieldOptional = MoreIterators.toList(fieldDefinition.get(ANY_OF_FIELD).elements()).stream().findFirst();
```

Comment:
This method always uses the first definition in the `anyOf` or `allOf` array. I think this approach won't work. If the value of the field does not follow the first definition, that value cannot be properly processed. For example, if this is the schema:

```json
{
  "anyOf": [
    { "type": "integer" },
    { "type": "boolean" }
  ]
}
```

, and if the value is a `boolean` instead of an `integer`, bigquery will think that an invalid value is imported. Is that the case?

Reply:
I updated it. Blocks `anyOf` and `allOf` are now converted to objects that contain all the fields from blocks `anyOf` and `allOf`.